lib/rust/ensogl/core/src/display/camera/camera2d.rs

```diff
@@ -415,7 +416,9 @@ impl Camera2d {
 impl Camera2d {
     /// Sets screen dimensions.
     pub fn set_screen(&self, width: f32, height: f32) {
-        self.data.borrow_mut().set_screen(width, height)
+        self.data.borrow_mut().set_screen(width, height);
+        let screen = self.data.borrow().screen;
+        self.data.borrow().frp.screen.emit(screen);
```

Comment:
I think we still could have the problem with double borrow - the `data` structure is borrowed while emitting `screen` event. We should instead clone_ref the screen endpoint from borrowed data and then emit the event.

Also, we can do this with only one borrow, avoiding unnecessary borrow checking:
```rust
   let (endpoint, screen) = {
       let mut data = self.data.borrow_mut();
       data.set_screen(width, height);
       (data.frp.screen.clone_ref(), data.screen)
   } // etc
```

Reply:
Nice catch, fixed. I also added a small unit test that catches invalid implementations. (not all of them)